hadoop-ozone/ozone-recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/FileSizeCountTask.java

```diff
@@ -0,0 +1,241 @@
+    updateCountFromDB();
+
+    while (eventIterator.hasNext()) {
+      OMDBUpdateEvent<String, OmKeyInfo> omdbUpdateEvent = eventIterator.next();
+      String updatedKey = omdbUpdateEvent.getKey();
+      OmKeyInfo updatedValue = omdbUpdateEvent.getValue();
+
+      try{
+        switch (omdbUpdateEvent.getAction()) {
+        case PUT:
+          updateUpperBoundCount(updatedValue, "PUT");
+          break;
+
+        case DELETE:
+          updateUpperBoundCount(updatedValue, "DELETE");
+          break;
+
+        default: LOG.trace("Skipping DB update event : " + omdbUpdateEvent
+                  .getAction());
+        }
+      } catch (IOException e) {
+        LOG.error("Unexpected exception while updating key data : {} {}",
+                updatedKey, e.getMessage());
+        return new ImmutablePair<>(getTaskName(), false);
+      }
+      populateFileCountBySizeDB();
+    }
+    LOG.info("Completed a 'process' run of FileSizeCountTask.");
+    return new ImmutablePair<>(getTaskName(), true);
+  }
+
+  /**
+   * Calculate the bin index based on size of the Key.
+   * index is calculated as the number of right shifts
+   * needed until dataSize becomes zero.
+   *
+   * @param dataSize Size of the key.
+   * @return int bin index in upperBoundCount
+   */
+  int calculateBinIndex(long dataSize) {
+    int index = 0;
+    while(dataSize != 0) {
+      dataSize >>= 1;
+      index += 1;
+    }
+    // The smallest file size being tracked for count
+    // is 1 KB i.e. 1024 = 2 ^ 10.
+    return index < 10 ? 0 : index - 10;
+  }
+
+  void countFileSize(OmKeyInfo omKeyInfo) {
+    int index;
+    if (omKeyInfo.getDataSize() >= maxFileSizeUpperBound) {
+      index = maxBinSize - 1;
+    } else {
+      index = calculateBinIndex(omKeyInfo.getDataSize());
+    }
+    upperBoundCount[index]++;
+  }
+
+  /**
+   * Populate DB with the counts of file sizes calculated
+   * using the dao.
+   *
+   */
+  void populateFileCountBySizeDB() {
+    for (int i = 0; i < upperBoundCount.length; i++) {
+      long fileSizeUpperBound = (i == upperBoundCount.length - 1) ?
+          Long.MAX_VALUE : (long) Math.pow(2, (10 + i));
+      FileCountBySize fileCountRecord =
+          fileCountBySizeDao.findById(fileSizeUpperBound);
+      FileCountBySize newRecord = new
+          FileCountBySize(fileSizeUpperBound, upperBoundCount[i]);
+      if (fileCountRecord == null) {
+        fileCountBySizeDao.insert(newRecord);
+      } else {
+        fileCountBySizeDao.update(newRecord);
+      }
+    }
+  }
+
+  private void updateUpperBoundCount(OmKeyInfo value, String operation)
+      throws IOException {
+    int binIndex = calculateBinIndex(value.getDataSize());
+    if (operation.equals("PUT")) {
+      upperBoundCount[binIndex]++;
+    } else if (operation.equals("DELETE")) {
+      if (upperBoundCount[binIndex] != 0) {
+        //decrement only if it had files before, default DB value is 0
+        upperBoundCount[binIndex]--;
```

Comment:
Same as above comment.

Reply:
handled this as part of clculateBinIndex()